soc/arm/nordic_nrf/nrf54l/soc.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for Nordic Semiconductor nRF54L family processor
+ *
+ * This module provides routines to initialize and support board-level hardware
+ * for the Nordic Semiconductor nRF54L family processor.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <cmsis_core.h>
+#include <soc/nrfx_coredep.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/cache.h>
+
+#include <hal/nrf_glitchdet.h>
+#include <hal/nrf_oscillators.h>
+#include <hal/nrf_power.h>
+#include <hal/nrf_regulators.h>
+
+#if defined(CONFIG_SOC_NRF54L15)
+#include <system_nrf54l.h>
+#else
+#error "Unknown nRF54L SoC."
+#endif
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+static int nordicsemi_nrf54l_init(void)
+{
+	/* Update the SystemCoreClock global variable with current core clock
+	 * retrieved from hardware state.
+	 */
+	SystemCoreClockUpdate();
+
+	/* Enable ICACHE */
+	sys_cache_instr_enable();
```

Comment:
do we have instruction to enable dcache?
I recall it was not present in the past and even if kconfig option was enabled, nothing called `sys_cache_data_enable`.

Reply:
nRF54L doesn't have DCACHE.